packages/astro/src/core/build/plugins/plugin-css.ts

```diff
@@ -243,8 +248,21 @@ function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin[] {
 							pageData.propagatedStyles.set(pageInfoId, new Set()).get(pageInfoId)!;
 
 						propagatedStyles.add(sheet);
+						sheetAddedToPage = true;
 					});
 				});
+
+				if (toBeInlined && sheetAddedToPage) {
+					// CSS is already added to all used pages, we can delete it from the bundle
+					// and make sure no chunks reference it via `importedCss` (for Vite preloading)
+					// to avoid duplicate CSS.
+					delete bundle[id];
```

Comment:
It's better to avoid `delete`
```js
bundle[id] = undefined
```
This page contains some information about why: https://biomejs.dev/linter/rules/no-delete

I predict that doing so will trigger the TS compiler, though

Reply:
The convention when dealing with rollup bundles is to `delete` it rather than setting it to undefined. Otherwise it kinda means that "the chunk exists but it's undefined".